homework5

danildemchenko opened the pull request.

danildemchenko committed:
homework5

VAmpersand commented on Module_2 Groupe_1/Danil/TabController/DanilTabViewController.swift:
```diff
@@ -10,11 +10,27 @@ import UIKit
         view.backgroundColor = .orange
+        addSubviews()
+        addConstraints()
+        transitionDemoButton.addTarget(self, action: #selector(transitionDemoButtonHandler), for: .touchUpInside)
```
Эту строку и 'view.backgroundColor_ = .orange' стоит убрать в configureAppearance

VAmpersand commented on Module_2 Groupe_1/Danil/TabController/DanilTabViewController.swift:
```diff
@@ -10,11 +10,27 @@ import UIKit
+    }
+    
+    private func addSubviews() {
+        view.addSubview(transitionDemoButton)
```
Ты не подтягивал main? Судя по всему нет. Подтяни его и пушни

VAmpersand commented on Module_2 Groupe_1/Danil/TabController/DetailsController.swift:
```diff
@@ -0,0 +1,17 @@
+import UIKit
+
+class DetailsController: UIViewController {
+
```
Если есть желание засунь сюда кнопку Close. И проверяя свойство контроллера isBeingPresented можешь обрабатывать выход или возвращение назад(для root можешь проверить наличие NavigationController и переставить его снова на табБар, как в видео). Только если есть желание

VAmpersand commented on Module_2 Groupe_1/Danil/PersonalHelpers/UIViewController + ext.swift:
```diff
@@ -18,5 +18,8 @@ extension UIViewController {
         static let middleMultipler = 0.5
         static let doubleMultiplier = 2
         static let divisor = 4
```
Я только сейчас заметил, как ты реализовал Constants. Не лучшее решение так делать. Во первых не надо делать этот, энум как  расширение UIViewController, он не нужен всем контроллерам. А если мы будем делать кастомные view то не дотянемся до них. Вынеси Constants в пространство проекта. И оставь в нем только глобальные значения. Типа mainOffset = 16 и тд. А уже в каждом конкретном файле делай узконеобоходимые для файла константы: 

import UIKit

private extension Constants {
    static let padding: CGFloat = 16
}

final class DanilTabViewController: UIViewController {
...

VAmpersand commented on Module_2 Groupe_1/Danil/TabController/TransitionDemoController.swift:
```diff
@@ -0,0 +1,109 @@
+    
+    @objc func seguePopoverButtonHandler() {
+        let detailsController = DetailsController()
+        detailsController.modalPresentationStyle = .popover
```
popover по умолчанию. Можно удалить

danildemchenko committed:
add name to files/classes

danildemchenko committed:
Merge branch 'main' into DanilsBranch

danildemchenko committed:
homework fixes

1. add configure method to UIViewControllers
2. implement Close button
3. rework Contants

VAmpersand commented on Module_2 Groupe_1/Danil/TabController/DanilDetailsController.swift:
```diff
@@ -0,0 +1,57 @@
+
+final class DanilDetailsController: UIViewController {
+    
+    private var isModal = false
```
Вопрос зачем делать еще одну переменную, когда у контроллера уже есть isBeingPresented. Можно смело его использовать вместо isModal ниже и удалить viewWillAppear.

danildemchenko replied:
у меня isBeingPresented отрабатывает корректно только в viewWillAppear
я пробовал с isBeingPresented работать с функции closeButtonHandler, значение false там

на скрине -- принчу значение isBeingPresented при нажатии на кнопку Close со всех трех нужных нам кейсов Segue
<img width="1245" alt="isBeing" src="https://user-images.githubusercontent.com/111361580/187967838-8b00472c-337c-44dd-af85-8a79613c2545.png">

VAmpersand replied:
Да, сам заметил на лекции. Что меня удивило, логика по моему не логичная)

VAmpersand commented on Module_2 Groupe_1/Danil/TabController/DanilDetailsController.swift:
```diff
@@ -0,0 +1,57 @@
+        } else if navigationController != nil {
+            navigationController?.popViewController(animated: true)
+        } else {
+            sceneDelegate?.window?.rootViewController = TabBarController()
```
Можно слегка улучшить этот код:
let tabBar = TabBarController()
tabBar.selectedIndex = Tabs.danil.rawValue
sceneDelegate?.window?.rootViewController = tabBar

danildemchenko replied:
доработал
